src/highlight.rs

```diff
@@ -37,89 +37,83 @@ impl Default for PickHighlightParams {
 pub struct HighlightablePickMesh {
     // Stores the initial color of the mesh material prior to selecting/hovering
     initial_color: Option<Color>,
+    // The pick group to use for determining highlight state
+    group: Group,
 }
 
 impl HighlightablePickMesh {
-    pub fn new() -> Self {
-        HighlightablePickMesh::default()
+    pub fn new(group: Group) -> Self {
+        HighlightablePickMesh {
+            group,
+            ..Default::default()
+        }
     }
 }
 
 impl Default for HighlightablePickMesh {
     fn default() -> Self {
         HighlightablePickMesh {
             initial_color: None,
+            group: Group::default(),
         }
     }
 }
 
-/// Given the current selected and hovered meshes and provided materials, update the meshes with the
-/// appropriate materials...
+/// Applies highlight an selection color to HighlightablePickMesh entities. Uses the group specified
+/// in the component.
 pub fn pick_highlighting(
     // Resources
-    pick_state: Res<PickState>,
     mut materials: ResMut<Assets<StandardMaterial>>,
     highlight_params: Res<PickHighlightParams>,
     // Queries
-    mut query_picked: Query<(
-        &mut HighlightablePickMesh,
-        &PickableMesh,
-        &Handle<StandardMaterial>,
-        Entity,
-    )>,
```

Comment:
Ah yes I see here, you don't get the conflicting query cause you've rolled it into a single one 🙂

Reply:
I wish I had noticed and saved you the time, I've been working on this sporadically and it's hard to keep track of what has diverged from master. 😅

Reply:
Haha no worries, looking at this mod has been really useful in getting a better idea for how bevy works / what it doesn't like 😂